Lib/configparser.py

```diff
@@ -850,11 +850,15 @@ def items(self, section=_UNSET, raw=False, vars=None):
         if vars:
             for key, value in vars.items():
                 d[self.optionxform(key)] = value
+        # Change `vars` from None to a dict to avoid TypeError on return.
+        else:
+            vars = {}
         value_getter = lambda option: self._interpolation.before_get(self,
             section, option, d[option], d)
         if raw:
             value_getter = lambda option: d[option]
-        return [(option, value_getter(option)) for option in d.keys()]
+        return [(option, value_getter(option)) for option in d.keys()
+                if option not in vars]
```

Comment:
If `vars` actually overrides existing defaults then your patch makes the original options disappear entirely. In fact, this creates a problem:
* if an existing option is shadowed by an entry in `vars`, should `items()` return the *original* value or the *overriden* value?

Reply:
I'm leaning towards answering the question above with "overriden value".  That would arguably be consistent with the docstring which states "Additional substitutions may be provided using the `vars' argument, which must be a dictionary **whose contents override any pre-existing defaults**."

Assuming this answer makes sense, the fix for the issue is in fact even shorter than your current patch:
* just add `orig_keys = list(d.keys())` after line 848
* use those `orig_keys` when iterating at `return` time

You don't have to change vars to `None`, and you don't have to filter the list.

Reply:
Thanks, that's a much nicer solution than mine.

Testing this has led to me discovering another discrepancy with the documentation that may be worth fixing first, however. See below.